builtin/providers/aws/resource_aws_iam_role.go

```diff
@@ -128,6 +139,13 @@ func resourceAwsIamRoleReadResult(d *schema.ResourceData, role *iam.Role) error
 	if err := d.Set("unique_id", role.RoleId); err != nil {
 		return err
 	}
+	policy, err := url.QueryUnescape(*role.AssumeRolePolicyDocument)
```

Comment:
I tried to do this using just the built-in `normalizeJson` function but receive the following error:

```
assume_role_policy: "Error parsing JSON: invalid character '%' looking for beginning of value" => "{\"Statement\":[{\"Action\":\"sts:AssumeRole\",\"Effect\":\"Allow\",\"Principal\":{\"AWS\":[\"arn:aws:iam::000000000000:role/ec2-rundeck\",\"arn:aws:iam::000000000000:role/ec2_rundeck\"],\"Service\":\"ec2.amazonaws.com\"},\"Sid\":\"\"}],\"Version\":\"2012-10-17\"}"
```

The policy returned in the underlying response is full of HTML character codes, like so:

```
%7B%22Version%22%3A%222012-10-17%22%2C%22Statement%22%3A%5B%7B%22Sid%22%3A%22%22%2C%22Effect%22%3A%22Allow%22%2C%22Principal%22%3A%7B%22Service%22%3A%22ec2.amazonaws.com%22%2C%22AWS%22%3A%5B%22arn%3Aaws%3Aiam%3A%3A00000000000%3Arole%2Fec2-rundeck%22%2C%22arn%3Aaws%3Aiam%3A%3A00000000000%3Arole%2Fec2_rundeck%22%5D%7D%2C%22Action%2
2%3A%22sts%3AAssumeRole%22%7D%5D%7D
```

In my latest commit it works nicely with the `url.QueryUnescape` in place. Should I be doing something else with the string before passing it to the `normalizeJson` function? Should the `url.QueryUnescape` be added in the `normalizeJson` function?

I looked at another resource (`resource_aws_cloudformation_stack.go`) where normalizeJson is used and compared the sample responses provided in the AWS API doco, and it doesn't seem to me as though the response should be so much different, so I will have to dig further to work out what the difference is.

The sample responses that I'm looking at are:
http://docs.aws.amazon.com/IAM/latest/APIReference/API_GetRole.html
http://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_GetTemplate.html